src/controllers/user/get-user.js

```diff
@@ -1,8 +1,23 @@
 const User = require('../../models/user');
+const constants = require('../../constants');
 
-const getUser = async (email) => {
+const {
+  AUTH__EMAIL_DOES_NOT_EXIST,
+  PASSWORD_RESET_LINK_INVALID_OR_EXPIRED,
+} = constants.errorCodes;
+
+const getUser = async (...fields) => {
```

Comment:
Logic to find the user from the database should be **fully** abstracted to this controller.  Here you're simply passing in all the fields and only mapping error code logic if `User.getUser` fails. This defeats the purpose of abstraction, which is to move re-used (or potentially re-used) implementation logic to a central place. Doing so also allows you to easily update the database layer without having to refactor the implementation of all the modules that use this abstraction.

Reply:
Sorry, I am been having a hard time trying to understand what you are suggesting here. 

Based on my understanding, are you suggesting that I should move the entire logic for getting the user (from this controller) to https://github.com/AntonioLok/online-shopping-reactJS-Backend/blob/master/src/models/user.js#L24?

Reply:
What you've created here is, in essence, a wrapper for `User.getUser`. It's a function that takes an arbitrary number of fields and queries the user object based on them. This heavily relies on the implementation of Mongo/Mongoose as is not really an abstraction but instead just a wrapper to handle/throw certain error codes.

Instead:
1. think about your use cases
2. create specific functions for each of them
3. if possible, find ways to combine those functions to make them reusable